src/node/createSetupServer.ts

```diff
@@ -103,16 +103,16 @@ export function createSetupServer(...interceptors: Interceptor[]) {
 
         if (!handler) {
           emitter.emit('request:unhandled', mockedRequest)
-        }
-
-        if (!response) {
-          emitter.emit('request:end', mockedRequest)
 
           onUnhandledRequest(
```

Comment:
I don't like how the worker and the server handle the same request/response flow in separate places. This opens a room to bugs, just as this one. It would be good to align this post-resolve handling into some sort of reused chain, the same for the worker and the server.